cep-recipe-jinja.md

```diff
@@ -0,0 +1,125 @@
+- `exact`: This is a boolean that specifies whether the pin should be exact. It defaults to `False`. If `exact` is `True`, the `min_pin` and `max_pin` are irrelevant. We pin to an `==` version and also include the build string exactly (e.g. `==1.2.3=h1234`).
+
+#### Example
+
+If we consider a package like `numpy-1.21.3-h123456_5` we could apply some pin expressions. 
+
+- `min_pin=x.x, max_pin=x.x` would result in `>=1.21,<1.22`
+- `min_pin=x.x.x,max_pin=x` would result in `>=1.21.3,<2`
+- `exact=True` would result in `==1.21.3=h123456_5`
+
+
+## The `pin_compatible` function
+
+Pin compatible will pin the dependency to the same version as "previously" resolved in the `host` or `build` environment. This is useful to ensure that the same package is used at run time as was used at build time.
+
+Example:
+
+```yaml
+requirements:
+  host:
+    - numpy
+  run:
+    - ${{ pin_compatible('numpy', exact=True) }}
+```
+
+## The `pin_subpackage` function
+
+Pin subpackage will pin the dependency to the same version as another sub-package from the recipe (or the current package itself).
+This is useful to ensure that multiple outputs from a recipe are linked together or to export the correct `run_exports` for a package.
+
+Example:
+
+```yaml
+outputs:
+  - package:
+      name: libfoo
+      version: "1.2.3"
+  - package:
+      name: foo
+      version: "1.2.3"
+    requirements:
+      run:
+        - ${{ pin_subpackage('libfoo', exact=True) }}
+```
+
+## The `cmp` function
+
+The `cmp` function is used to compare two versions. It returns `True` if the comparison is true and `False` otherwise.
+
+## The `hash` variable
+
+`${{ hash }}` is the variant hash and is useful in the build string computation. This used to be `PKG_HASH` in the old recipe format. Since the `hash` variable depends on the variant computation, it is only available in the `build.string` field and is computed after the entire variant computation is finished.
+
+## The `version_to_buildstring` function
+
+- `${{ python | version_to_buildstring }}` converts a version from the variant to a build string (it removes the `.` character and takes only the first two elements of the version).
+
+## The `env` object
+
+You can use the `env` object to retrieve environment variables and forward them to your build script. There are two ways to do this:
+
+- `${{ env.get("MY_ENV_VAR") }}` will return the value of the environment variable `MY_ENV_VAR` or throw an error if it is not set.
+- `${{ env.get_default("MY_ENV_VAR", "default_value") }}` will return the value of the environment variable `MY_ENV_VAR` or `"default_value"` if it is not set.
+
+You can also check for the existence of an environment variable:
+
+- `${{ env.exists("MY_ENV_VAR") }}` will return `true` if the environment variable `MY_ENV_VAR` is set and `false` otherwise.
```

Comment:
Nitpick. `false` and `False` are both used in the spec. It's hard to know if it returns a string or a bool.

Reply:
I added the word `boolean true` to make it clear that this is a bool value.